### ISSUE TITLE
Installation on Mac OSX 10.6 (snow leopard) and MacPorts Python 2.4

### ISSUE THREAD

**Issue by GoogleCodeExporter** [maximemaher/pyodbc#105]
```
Hi,

I got a compilation issue on the above mentioned platform. See the attached 
installation log.

Regards and many thanks for your support.

-- 
Gilles Lenfant
```

Original issue reported on code.google.com by `gilles.l...@gmail.com` on 25 Jun 2010 at 12:34

Attachments:
- [easy_install.log](https://storage.googleapis.com/google-code-attachments/pyodbc/issue-105/comment-0/easy_install.log)

**Comment by GoogleCodeExporter** [comment]
```
Oops. Forgot to mention: The installation fails with Python 2.4 but is OK with 
Python 2.6

```

Original comment by `gilles.l...@gmail.com` on 25 Jun 2010 at 12:38

**Comment by GoogleCodeExporter** [comment]
```
If possible, can you test this with the v2unicode branch?

Most of the Linux & OS/X issues were due to incorrect handling of UCS4 which 
I've corrected on that branch.  As soon as I'm sure it has been tested enough, 
it will become 2.1.8.

```

Original comment by `mkleehammer` on 5 Sep 2010 at 6:18
- Changed state: **Investigating**

**Comment by GoogleCodeExporter** [comment]
```
Can you test this with 2.1.8?

Thanks.
```

Original comment by `mkleehammer` on 6 Sep 2010 at 5:36

**Comment by GoogleCodeExporter** [comment]
```
Sorry to answer late (coming back from vacations). I tried again and got this 
with 2.1.8 (other errors).

The full console log is in attachment of this comment.

Thank you again for your support.

```

Original comment by `gilles.l...@gmail.com` on 8 Sep 2010 at 1:42

Attachments:
- [pyodbc-error-compile.txt](https://storage.googleapis.com/google-code-attachments/pyodbc/issue-105/comment-4/pyodbc-error-compile.txt)

**Comment by GoogleCodeExporter** [comment]
```
OK, it looks like Python 4 is not using "const char*" where it should.  Most of 
the time a string literal is used which results in the warning "deprecated 
conversion from string constant to ‘char*’" and the compiler performs the 
cast for us.  However, the one time I pass a "const char*" variable, the 
compiler won't (and shouldn't), so there is an error.

My first question is, is Python 2.4 the version that comes with OS/X?  I do 
support 2.4, but am wondering how common this will be.

Second, I'll make the appropriate cast to "char*" on that line and make 2.1.9.  
I guess I'll have to buy a Mac :)
```

Original comment by `mkleehammer` on 8 Sep 2010 at 2:26
- Added labels: **OpSys-OSX**

**Comment by GoogleCodeExporter** [comment]
```
What a fast answer... ;)

I'm familiar with this deprecation warning that does not prevent having C 
extensions that work on Mac. The real errors are stated in lines 226 and 227 of 
my last attachment.
<quoting>
.../pyodbc-2.1.8/src/pyodbcmodule.cpp:919: error: invalid conversion from 
‘const char*’ to ‘char*’
.../pyodbc-2.1.8/src/pyodbcmodule.cpp:919: error:   initializing argument 3 of 
‘int PyModule_AddStringConstant(PyObject*, char*, char*)’
</quoting>

I'm using Python 2.4 installed with MacPorts that works perfectly with lots of 
complex Python apps (Zope, ...) and C based eggs (python-ldap, lxml, ...)

http://trac.macports.org/browser/trunk/dports/lang/python24

The default compiler is gcc-4.2.

 If you need more investigation I could give you an ssh access to my macbook.


```

Original comment by `gilles.l...@gmail.com` on 8 Sep 2010 at 2:51

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
FYI: I am building for CentOS 5.4 and I get the same error in pyodbcmodule.cpp 
at line 919.

OS: CentOS 5.4
Python: 2.4.3
gcc: 4.1.2

Thanks.
```

Original comment by `dccar...@gmail.com` on 29 Sep 2010 at 10:00

**Comment by GoogleCodeExporter** [comment]
```
Same here, it fails to compile with:

Python 2.4.3
CentOS release 5.5 (Final)
gcc (GCC) 4.1.2 20080704 (Red Hat 4.1.2-48)

I compiled a newer Python and it worked:
Python 2.6.6
```

Original comment by `jean.jor...@gmail.com` on 15 Nov 2010 at 5:27

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 2.1.9-beta06

```

Original comment by `mkleehammer` on 21 Nov 2010 at 6:24
- Changed state: **Complete**

**Comment by GoogleCodeExporter** [comment]
```
May I ask to release such version, 2.1.9? I'm using 2.4 with older version of 
zope. I fixed the bug manually in 2.1.8, but it would be convenient to have an 
official package.
```

Original comment by `fcormart...@gmail.com` on 20 Mar 2011 at 9:27
